provider/pkg/gen/dotnet.go

```diff
@@ -17,56 +17,82 @@ package gen
 
 import (
 	"fmt"
+
+	"github.com/pulumi/pulumi/pkg/v3/codegen/cgstrings"
 )
 
-var pascalCaseMapping = map[string]string{
-	"admissionregistration": "AdmissionRegistration",
-	"apps":                  "Apps",
-	"auditregistration":     "AuditRegistraion",
-	"authentication":        "Authentication",
-	"apiextensions":         "ApiExtensions",
-	"authorization":         "Authorization",
-	"autoscaling":           "Autoscaling",
-	"apiregistration":       "ApiRegistration",
-	"batch":                 "Batch",
-	"certificates":          "Certificates",
-	"coordination":          "Coordination",
-	"core":                  "Core",
-	"discovery":             "Discovery",
-	"events":                "Events",
-	"extensions":            "Extensions",
-	"flowcontrol":           "FlowControl",
-	"networking":            "Networking",
-	"meta":                  "Meta",
-	"node":                  "Node",
-	"policy":                "Policy",
-	"rbac":                  "Rbac",
-	"resource":              "Resource",
-	"scheduling":            "Scheduling",
-	"settings":              "Settings",
-	"storage":               "Storage",
-	"storagemigration":      "StorageMigration",
-	"v1":                    "V1",
-	"v1alpha1":              "V1Alpha1",
-	"v1alpha2":              "V1Alpha2",
-	"v1alpha3":              "V1Alpha3",
-	"v1beta1":               "V1Beta1",
-	"v1beta2":               "V1Beta2",
-	"v1beta3":               "V1Beta3",
-	"v2":                    "V2",
-	"v2alpha1":              "V2Alpha1",
-	"v2beta1":               "V2Beta1",
-	"v2beta2":               "V2Beta2",
+// CaseMapping is a mapping of names to their PascalCase equivalents. This is a
+// a data structure that only allows adding new mappings, and will error if a
+// mapping already exists.
+type CaseMapping struct {
+	mapping map[string]string
+}
+
+func (c *CaseMapping) Add(name, pascal string) error {
+	if _, ok := c.mapping[name]; ok {
+		return fmt.Errorf("case mapping for %q already exists", name)
+	}
+
+	// Ensure the pascal case name is actually pascal case, otherwise capitalize the first letter.
```

Comment:
Updated to use a forked copy (with tests) of Pulumi's `cgstrings.ModifyStringAroundDelimeter` which accepts a regex delimiter instead of a fixed string for this purpose.